apps/designer/app/designer/features/style-panel/controls/text/text-control.tsx

```diff
@@ -0,0 +1,128 @@
+import { ChangeEvent } from "react";
+import {
+  Flex,
+  Grid,
+  Combobox,
+  IconButton,
+  TextField,
+  Tooltip,
+} from "@webstudio-is/design-system";
+import { PropertyName } from "../../shared/property-name";
+import { getFinalValue } from "../../shared/get-final-value";
+import { useIsFromCurrentBreakpoint } from "../../shared/use-is-from-current-breakpoint";
+import { ControlProps } from "../../style-sections";
+import { RowGapIcon, ColumnGapIcon } from "@webstudio-is/icons";
+import { Unit } from "../../shared/unit";
+
+export const TextControl = ({
```

Comment:
I don't understand what this control is for from its semantics. It also seems like it could be 2 separate controls, because renders an entirely different interface depending on property and it shouldn't care about the property at all since we extracted the mapping to outside